torch/autograd/saved_variable_default_hooks.py

```diff
@@ -8,18 +8,49 @@ def reset_saved_tensors_default_hooks():
     torch._C._autograd._reset_default_hooks()
 
 class save_on_cpu(object):
+    """"Context-manager under which tensors saved by the forward pass will be
+    stored on cpu, then retrieved for backward.
+
+    When performing operations within this context manager, intermediary
+    results saved in the graph during the forward pass will be moved to CPU,
```

Comment:
What do you think about a quick pitch on why this feature is useful? Something like "this may be helpful if your model doesn't fit in GPU memory"